iib/workers/tasks/build_regenerate_bundle.py

```diff
@@ -0,0 +1,356 @@
+    ``metadata/annotations.yaml``.
+
+    The final package name value is returned as part of the tuple.
+
+    :param str metadata_path: the path to the bundle's metadata directory.
+    :param str organization: the organization this customization is for.
+    :raise IIBError: if the ``metadata/annotations.yaml`` file is in an unexpected format.
+    :return: a tuple with the package name and a dictionary of labels to set on the bundle.
+    :rtype: tuple(str, dict)
+    """
+    annotations_yaml_path = os.path.join(metadata_path, 'annotations.yaml')
+    if not os.path.exists(annotations_yaml_path):
+        raise IIBError('metadata/annotations.yaml does not exist in the bundle')
+
+    with open(annotations_yaml_path, 'r') as f:
+        try:
+            annotations_yaml = yaml.load(f)
+        except ruamel.yaml.YAMLError:
+            error = 'metadata/annotations/yaml is not valid YAML'
+            log.exception(error)
+            raise IIBError(error)
+
+    if not isinstance(annotations_yaml.get('annotations', {}), dict):
+        raise IIBError('The value of metadata/annotations.yaml must be a dictionary')
+
+    package_label = 'operators.operatorframework.io.bundle.package.v1'
+    package_annotation = annotations_yaml.get('annotations', {}).get(package_label)
+    if not package_annotation:
+        raise IIBError(f'{package_label} is not set in metadata/annotations.yaml')
+
+    if not isinstance(package_annotation, str):
+        raise IIBError(f'The value of {package_label} in metadata/annotations.yaml is not a string')
+
+    if not organization:
+        log.debug('No organization was provided to add the package name suffix')
+        return package_annotation, {}
+
+    conf = get_worker_config()
+    package_name_suffix = (
+        conf['iib_organization_customizations'].get(organization, {}).get('package_name_suffix')
+    )
+    if not package_name_suffix:
+        log.debug(
+            'The "package_name_suffix" configuration is not set for the organization %s',
+            organization,
+        )
+        return package_annotation, {}
+
+    if package_annotation.endswith(package_name_suffix):
+        log.debug('No modifications are needed on %s in metadata/annotations.yaml', package_label)
+        return package_annotation, {}
+
+    annotations_yaml['annotations'][package_label] = f'{package_annotation}{package_name_suffix}'
+
+    with open(annotations_yaml_path, 'w') as f:
+        yaml.dump(annotations_yaml, f)
+
+    log.info(
+        'Modified %s in metadata/annotations.yaml from %s to %s',
+        package_label,
+        package_annotation,
+        annotations_yaml['annotations'][package_label],
+    )
+
+    return (
+        annotations_yaml['annotations'][package_label],
+        {package_label: annotations_yaml['annotations'][package_label]},
+    )
+
+
+def _adjust_operator_bundle(manifests_path, metadata_path, organization=None, pinned_by_iib=False):
```

Comment:
Do we want to get rid of `_` at the beginning of the name of the function?

Reply:
I don't think so because it's still a private function being used only in the same module.